src/main/java/teammates/ui/template/AdminActivityLogTableRow.java

```diff
@@ -0,0 +1,211 @@
+    public String getDisplayedLogTime() {
+        Calendar appCal = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        SimpleDateFormat sdf = new SimpleDateFormat("dd-MM-yyyy HH:mm:ss");
+        sdf.setTimeZone(TimeZone.getTimeZone(Const.SystemParams.ADMIN_TIME_ZONE));
+        appCal.setTimeInMillis(activityLog.getTime());
+    
+        return sdf.format(appCal.getTime());
+    }
+
+    public String getDisplayedRole() {
+        return activityLog.getRole(); // will change later
+    }
+
+    public String getDisplayedLogTimeTaken() {
+        return TimeHelper.convertToStandardDuration(activityLog.getTimeTaken());
+    }
+
+    public String getDisplayedMessage() {
+        String displayedMessage = activityLog.getMessage();
+        String[] keywords = {
+                Const.ACTION_RESULT_FAILURE,
+                Const.ACTION_RESULT_SYSTEM_ERROR_REPORT
+        };
+        displayedMessage = highlightKeyword(displayedMessage, keywords,
+                ACTION_UNSCCUSSFUL_HIGHLIGHTER_FRONT, ACTION_UNSCCUSSFUL_HIGHLIGHTER_BACK);
+        displayedMessage = highlightKeyword(displayedMessage, keyStringsToHighlight,
+                KEYWORDS_HIGHLIGHTER_FRONT, KEYWORDS_HIGHLIGHTER_BACK);
+        return displayedMessage;
+    }
+    
+    // --------------- Forwarding activityLog methods ---------------
+    
+    public String getUserGoogleId() {
+        return activityLog.getGoogleId();
+    }
+
+    public String getUserName() {
+        return activityLog.getName();
+    }
+
+    public String getUserEmail() {
+        return activityLog.getEmail();
+    }
+
+    public String getLogId() {
+        return activityLog.getId();
+    }
+
+    public String getActionName() {
+        return activityLog.getAction();
+    }
+
+    public String getLogTime() {
+        return String.valueOf(activityLog.getTime());
+    }
+    
+    public boolean getIsMasqueradeUserRole() {
+        return activityLog.isMasqueradeUserRole();
+    }
+    
+    // --------------- Helper methods ---------------
+    
+    // TODO: can be generalized to helper method
+    private String highlightKeyword(String original, String[] keywords,
+                                    String wrapperTagFront, String wrapperTagEnd) {
+        
+        if (keywords == null) {
+            return original;
+        }
+        
+        String highlightedString = original;
+        
+        for (String stringToHighlight : keywords) {
+            if (highlightedString.toLowerCase().contains(stringToHighlight.toLowerCase())) {
+                
+                int startIndex = original.toLowerCase().indexOf(stringToHighlight.toLowerCase());
+                int endIndex = startIndex + stringToHighlight.length();
+                String realStringToHighlight = original.substring(startIndex, endIndex);
+                highlightedString = highlightedString.replace(realStringToHighlight,
+                        wrapperTagFront + realStringToHighlight + wrapperTagEnd);
```

Comment:
From what I understand, this is attempting to do `a.replace(b, c)` but case insensitive?
In that case, [regex replace](http://stackoverflow.com/questions/11236610/case-insensitive-variable-for-string-replaceall-method-java) might be your friend.